lib/asyn-ares.c

```diff
@@ -707,6 +707,11 @@ CURLcode Curl_set_dns_servers(struct Curl_easy *data,
   CURLcode result = CURLE_NOT_BUILT_IN;
   int ares_result;
 
+  if(data->dns_servers) {
```

Comment:
Could I use Curl_safefree here? I notice other code in this file uses plain free.

Reply:
Yes you can use Curl_safefree, and you actually don't have to check the pointer first, as it can deal with NULL there just fine..